meson.build

```diff
@@ -322,28 +331,28 @@ if not meson.is_subproject()
   subdir('binr/rax2')
 else
   libr2_dep = declare_dependency(
```

Comment:
Replace it with array of dependencies

Reply:
why? isn't it better this way, since you also define the `include_directories` and the `version`?

Reply:
Each `r_*_dep` dependency already has `include_directories`. And `version` should be defined in each `r_*_dep` too.

Reply:
But that would behave differently. Currently you can consider `libr2_dep` a dependency object (e.g. you can do `libr2_dep.found()`, `libr2_dep.version`, etc.), but if you convert this into an array you don't have all those goodies anymore.

You would have to consider each element separately and not as a whole single library.

Reply:
`found()` dependency_object's method is usefull only if you use `dependency()` function. Pkg-config files are generated for each separate r_* library, not for whole libr2.
http://mesonbuild.com/Reference-manual.html#dependency

Reply:
`dependency` or `declare_dependency`. If r2 is used as a subproject of another project, you can do something like this (as explained in http://mesonbuild.com/Subprojects.html):
```
radare2_sp = subproject('radare2')
libr2_dep = radare2_sp.get_variable('libr2_dep')
```
It's true that right now you don't have the pkg-config file that would allow you to use `dependency` (in the default way), but I see only possible advantanges in `declare_dependency` instead of just using an array.

Reply:
It is not critical so you can merge it now. But I still think that declaring libr2 dependency is unnecessarily.